examples/tinywl-new/surfacewrapper.cpp

```diff
@@ -476,18 +519,40 @@ void SurfaceWrapper::requestMinimize()
     setSurfaceState(State::Minimized);
 }
 
-void SurfaceWrapper::requestUnminimize()
+void SurfaceWrapper::requestCancelMinimize()
 {
     if (m_surfaceState == State::Minimized)
         setSurfaceState(m_previousSurfaceState);
 }
 
+void SurfaceWrapper::requestMaximize()
+{
+    setSurfaceState(State::Maximized);
+}
+
+void SurfaceWrapper::requestCancelMaximize()
+{
+    if (m_surfaceState == State::Maximized)
+        setSurfaceState(m_previousSurfaceState);
+}
+
 void SurfaceWrapper::requestToggleMaximize()
 {
     if (m_surfaceState == State::Maximized)
-        setSurfaceState(State::Normal);
+        requestCancelMaximize();
```

Comment:
setSurfaceState(State::Normal)

Reply:
set `Normal` even though previous state is `Tiling` ?

Reply:
The `maximize` state can't cancel, we not need a `requestCancelMaximize` function, it like as `Normal` state, shouldn't cancel normal state and restore the previous state.

Reply:
A surface request maximize from minimized state, and user click titlebar's button to toggle maximize, if you `requestCancelMaximize` here, this surface will be minimize state.